html/modules/custom/bc_dc/src/Controller/BcDcCreateFileController.php

```diff
@@ -4,105 +4,105 @@
-              foreach ($worksheet->getColumnIterator() as $column) {
-                $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(TRUE);
-              }
+      $spreadsheet = new Spreadsheet();
+      switch ($param) {
+        case 'csv':
+          $filename = $path . '_ID_' . $nid . '.xlsx';
+          // Save to file.
+          $worksheets = new Worksheet($spreadsheet, "Sheet 1");
+          $spreadsheet->addSheet($worksheets, 0);
+          $worksheets->fromArray($sheetData);
+          foreach ($worksheets as $worksheet) {
+            foreach ($worksheet->getColumnIterator() as $column) {
+              $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(TRUE);
             }
-            $writer = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($spreadsheet);
-            $directory = 'public://data_dictionary/' . $filename;
-            $writer->save($directory);
+          }
+          $writer = new Xlsx($spreadsheet);
+          $directory = 'public://data_dictionary/' . $filename;
+          $writer->save($directory);
 
-          case 'xlsx':
-            $filename = $path . '_ID_' . $nid . '.csv';
-            $mySpreadsheet = new Spreadsheet();
-            $worksheets = new \PhpOffice\PhpSpreadsheet\Worksheet\Worksheet($mySpreadsheet, "Sheet 1");
-            $mySpreadsheet->addSheet($worksheets, 0);
-            $worksheets->fromArray($sheetData);
-            foreach ($worksheets as $worksheet) {
-              foreach ($worksheet->getColumnIterator() as $column) {
-                $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(TRUE);
-              }
+        case 'xlsx':
+          $filename = $path . '_ID_' . $nid . '.csv';
+          $mySpreadsheet = new Spreadsheet();
+          $worksheets = new Worksheet($mySpreadsheet, "Sheet 1");
+          $mySpreadsheet->addSheet($worksheets, 0);
+          $worksheets->fromArray($sheetData);
+          foreach ($worksheets as $worksheet) {
+            foreach ($worksheet->getColumnIterator() as $column) {
+              $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(TRUE);
             }
-            $writer = new \PhpOffice\PhpSpreadsheet\Writer\Csv($mySpreadsheet);
-            $directory = 'public://data_dictionary/' . $filename;
-            $writer->setDelimiter(',');
-            $writer->setEnclosure('');
-            $writer->setLineEnding("\r\n");
-            $writer->setSheetIndex(0);
-            $writer->save($directory);
-        }
-    } else {
-        throw new NotFoundHttpException();
+          }
+          $writer = new Csv($mySpreadsheet);
+          $directory = 'public://data_dictionary/' . $filename;
+          $writer->setDelimiter(',');
+          $writer->setEnclosure('');
+          $writer->setLineEnding("\r\n");
+          $writer->setSheetIndex(0);
+          $writer->save($directory);
+      }
+    }
+    else {
+      throw new NotFoundHttpException();
```

Comment:
Move this to near the top of the function:

```
if (!in_array($param, $allowedExt, TRUE)) {
  throw new NotFoundHttpException();
}
```
The rest of the function would not be inside the `if` statement at all.